PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx

```diff
@@ -0,0 +1,399 @@
+
+  enum HFStep { kHFStepMC = 0,              // all generated mothers
+                kHFStepMcInRapidity,        // MC mother in rapidity |y| < rapidityCharmBaryonMax=0.5
+                kHFStepAcceptance,          // MC mother where all final state candidates pass eta and pt selection
+                kHFStepTrackable,           // MC mother where all final state candidates have a reconstructed track
+                kHFStepAcceptanceTrackable, // MC mother where all final state candidates have a reconstructed track and pass eta and pt selection
+                kHFStepItsTpcTrackableCuts, // MC mother where charm bachelor passes isGlobalTrkWoDca selection
+                kHFStepItsTrackableCuts,    // MC mother where charm bachelor passes isITSQualityTrack selection
+                kHFStepTracked,             // signal candidates which have been found by candidateCreator for the desired decay channel
+                kHFStepTrackedCuts,         // signal candidates which have been found and pass the candidateSelector cuts but PID
+                kHFStepTrackedSelected,     // signal candidates which pass the selector and pass the candidateSelector cuts including PID
+                kHFNSteps };
+
+  enum TrackableStep { kTrackableAll = 0, // all tracks
+                       kTrackableITS,     // track contains ITS information
+                       kTrackableTPC,     // track contains TPC information
+                       kTrackableITSTPC,  // track contains ITS and TPC information
+                       kNTrackableSteps };
+
+  using TracksWithSelectionMC = soa::Join<aod::Tracks, aod::TracksExtra, aod::McTrackLabels, aod::TrackSelection>;
+  using CandidateInfo = soa::Join<aod::HfCandToXiPi, aod::HfToXiPiMCRec, aod::HfSelToXiPi>;
+  using ParticleInfo = soa::Join<aod::McParticles, aod::HfToXiPiMCGen>;
+
+  HistogramRegistry registry{"registry"};
+
+  void init(InitContext&)
+  {
+    auto hCandidates = registry.add<StepTHn>("hCandidates", "Candidate count at different steps", {HistType::kStepTHnF, {axisPt, axisMass, {2, -0.5, 1.5, "collision matched"}, {RecoDecay::OriginType::NonPrompt + 1, RecoDecay::OriginType::None - 0.5, RecoDecay::OriginType::NonPrompt + 0.5}}, kHFNSteps});
+    hCandidates->GetAxis(0)->SetTitle("#it{p}_{T} (GeV/#it{c})");
+    hCandidates->GetAxis(1)->SetTitle("#it{m}_{inv} (GeV/#it{c}^{2})");
+    hCandidates->GetAxis(2)->SetTitle("Collision matched");
+    hCandidates->GetAxis(3)->SetTitle("Charm hadron origin");
+
+    auto hTrackablePtEta = registry.add<StepTHn>("hTrackablePtEta", "Prongs kinematics at different steps", {HistType::kStepTHnF, {{200, 0, 20}, {400, -2, 2}}, kNTrackableSteps});
+    hTrackablePtEta->GetAxis(0)->SetTitle("#it{p}_{T} (GeV/#it{c})");
+    hTrackablePtEta->GetAxis(1)->SetTitle("#eta");
+
+    if (doprocessOmegac0 == true && doprocessXic0 == true) {
+      LOGF(fatal, "Can't enable processOmegac0 and processXic0 at the same time. Please choose one.");
+    }
```

Comment:
- This should be the first check to do.
- ` == true` is redundant.
- You are not covering the case of no enabled function.

Reply:
In the end I removed this check, I switched to 1 process function only where the configurables allow to select which particle to match (and I moved a check on the configurable values inside the process function itself)

Reply:
Perfect! Thanks! Now it is much simpler and clearer.